### PR TITLE
Internal Twig Updates

### PR DESCRIPTION
**Pull request by japaveh**
To be able to use the software with the latest version of Twig some small (BC) patches are needed.
Since this fork is closest to ZF3 I created a fork of this one.

### CODE REVIEW

File: src/ZfcTwig/Twig/FallbackFunction.php [hunk 1]
```diff
@@ -5,7 +5,7 @@
 use Twig_Function;
```
Comment (kokspflanze) [inline src/ZfcTwig/Twig/FallbackFunction.php]: you can also change `Twig_Function` to `Twig_SimpleFunction` so we can also remove the `\` before the `Twig_SimpleFunction` class

File: src/ZfcTwig/Twig/StackLoader.php [hunk 2]
```diff
@@ -41,18 +41,24 @@ public function getDefaultSuffix()
...
+            }
+
+            throw new Twig_Error_Loader($this->errorCache[$name]);
         }
 
         // Ensure we have the expected file extension
         $defaultSuffix = $this->getDefaultSuffix();
-        if (pathinfo($name, PATHINFO_EXTENSION) != $defaultSuffix) {;
+        if (pathinfo($name, PATHINFO_EXTENSION) != $defaultSuffix) {
+            ;
```
Comment (kokspflanze) [inline src/ZfcTwig/Twig/StackLoader.php]: can you please remove the `;` too

File: src/ZfcTwig/Twig/StackLoader.php [hunk 3]
```diff
@@ -41,18 +41,24 @@ public function getDefaultSuffix()
...
     {
-        $name = (string) $name;
+        $throw = func_num_args() > 1 ? func_get_arg(1) : true;
+        $name  = (string)$name;
 
         // normalize name
         $name = preg_replace('#/{2,}#', '/', strtr($name, '\\', '/'));
 
         if (isset($this->cache[$name])) {
-            return $this->cache[$name];
```
Comment (kokspflanze) [inline src/ZfcTwig/Twig/StackLoader.php:50]: i dont realy get it, why throw an `Exception`, if the catch match?

i think it should be like https://github.com/twigphp/Twig/blob/1.x/lib/Twig/Loader/Filesystem.php#L203-L213

**Review by kokspflanze — CHANGES_REQUESTED** [review]
thanks, the important part is this https://github.com/kokspflanze/ZfcTwig/pull/3/files/1971d839036c08d3df497d35ac999763e9dc8a89#diff-1379b576bb8f6c4f41991ba2423983b6 can you give me a reason for this?

**Comment by japaveh** [comment]
The strange exception was a copy-paste error from Twig, my bad. I fixed it now

**Review by kokspflanze — APPROVED** [review]

**Comment by kokspflanze** [comment]
thanks, i will do some checks and add soon a new release.

**Pull request merged**